Validation/SiTrackerPhase2V/plugins/Phase2OTValidateTTStub.cc

```diff
@@ -38,96 +44,475 @@
 #include "Geometry/Records/interface/TrackerTopologyRcd.h"
 #include "Geometry/TrackerGeometryBuilder/interface/StripGeomDetUnit.h"
 #include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/CommonTopologies/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
 
 #include "DQMServices/Core/interface/DQMEDAnalyzer.h"
 #include "DQMServices/Core/interface/DQMStore.h"
 
 class Phase2OTValidateTTStub : public DQMEDAnalyzer {
 public:
-  explicit Phase2OTValidateTTStub(const edm::ParameterSet &);
+  explicit Phase2OTValidateTTStub(const edm::ParameterSet&);
   ~Phase2OTValidateTTStub() override;
-  void analyze(const edm::Event &, const edm::EventSetup &) override;
-  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
-  void dqmBeginRun(const edm::Run &iRun, const edm::EventSetup &iSetup) override;
-  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+  void dqmBeginRun(const edm::Run& iRun, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  float phiOverBendCorrection(bool isBarrel,
+                              float stub_z,
+                              float stub_r,
+                              const TrackerTopology* tTopo,
+                              uint32_t detid,
+                              const GeomDetUnit* det0,
+                              const GeomDetUnit* det1);
+  std::vector<double> getTPDerivedCoords(
+      edm::Ptr<TrackingParticle> my_tp, bool isBarrel, double modMaxZ, double modMinZ, float stub_r) const;
+
   // TTStub stacks
   // Global position of the stubs
-  MonitorElement *Stub_RZ = nullptr;  // TTStub #rho vs. z
+  MonitorElement* Stub_RZ = nullptr;  // TTStub #rho vs. z
+
+  // delta_z hists
+  MonitorElement* z_res_isPS_barrel;
+  MonitorElement* z_res_is2S_barrel;
+
+  // delta_phi hists
+  MonitorElement* phi_res_barrel;
+  MonitorElement* phi_res_barrel_L1;
```

Comment:
Instead of declaring separately phi_res_barrel_L1, phi_res_barrel_L2 etc., which requires many lines of code, use an std::vector<MonitorElement*>. Then when booking and filling the histograms, you can use for loops to make the code much shorter.